Dev/thirdparty

huan5678 opened the pull request:
更新 登入頁面的 第三方登入串接後端API
更新 NavBar 接第三方回來時的token並驗證，並完善swal的提示

huan5678 committed:
GIT 提交訊息：

功能：重構認證用戶介面元件

- 重構導航欄樣式，重新排列類別以提高可讀性
- 將導航欄中的佔位符 `/none` 链接替換為適當路由
- 刪除導航欄中未使用的類別並完善連結的懸停樣式
- 根據認證狀態動態顯示登錄或會員信息於導航欄
- 更新導航欄中的登出和會員中心按鈕行為
- 為導航欄元件新增登出和令牌驗證方法
- 在導航欄元件中實現掛載時檢查認證狀態的邏輯
- 重新排序並簡化登錄視圖中的模板元素類別
- 為登錄視圖表單按鈕新增加載狀態，防止多次提交
- 用動態方式替換登錄視圖中的靜態登錄方法，包括第三方 OAuth 選項
- 包括在登錄視圖中顯示錯誤訊息和在登錄失敗後清除表單的處理
- 在登錄視圖中新增同意隱私政策勾選框以及相應的驗證和錯誤訊息展示

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息:

重構: 重構身份驗證狀態處理

- 重构 `authCheck` 以使用 `checkUser` 方法来更新用户状态，进行身份验证后检查
- 移除 `authCheck` 方法中的控制台日志
- 确保在令牌存在和令牌cookie情况下，令牌验证后，通过 `checkUser` 更新用户登录状态

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

重構：為了清晰及整潔重構 `NavbarEl.vue`

- 更新 `NavbarEl.vue` 元件中成功登入訊息的格式
- 從 `NavbarEl.vue` 元件的 `authCheck` 方法中移除 console log 語句

Signed-off-by: huan5678 <huan5678@gmail.com>

echocarriet commented:
我這邊第三方登入的部分除了 Gmail 外，其他都顯示下方的無法登入的視窗，其他部分沒問題喔！
![截圖 2024-02-26 晚上8 53 59](https://github.com/sp1daTomm/SelectWave/assets/52949067/8423e2a4-fb1d-4cd4-b6e8-3f4f5e2a3d43)
![截圖 2024-02-26 晚上8 54 15](https://github.com/sp1daTomm/SelectWave/assets/52949067/108173ad-9887-4231-8714-f209595778e9)
![截圖 2024-02-26 晚上8 54 26](https://github.com/sp1daTomm/SelectWave/assets/52949067/c861ed5e-bb58-4d5d-854d-7831f101d190)
![截圖 2024-02-26 晚上8 54 37](https://github.com/sp1daTomm/SelectWave/assets/52949067/3f1e5d62-995e-47f5-8dd3-6c3d63db13f7)

eee317 commented:
我用第三方登入沒有成功XD

huan5678 commented:
第三方沒成功的大概應該是後端目前寫的前端位置不正確的緣故，所以最後一步轉回前端頁面沒有成功，我現在暫時把後端env對應前端的位置改成localhost:5173，讓他先對大家本地端的vite dev位置，大家可以再試試看了

huan5678 committed:
GIT 提交訊息:

特性：重構認證和狀態管理

- 重構 `App.vue` 腳本，以利用組合式 API 改善認證功能和用戶資料處理
- 在 `NavbarEl.vue` 中使用 Pinia 實現用戶認證和狀態管理
- 在 `main.js` 中添加 `VueAxios` 作為全局插件至應用程式
- 創建 `member.js` Pinia 商店，用於集中式會員狀態管理

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交信息：

功能：重構身份驗證 UI 流程和狀態處理

- 在 NavbarEl 中替換用戶圖標為動態頭像圖片，並註釋掉用戶名的顯示
- 將計算屬性從 'member' 重命名為 'memberStore'，並調整 NavbarEl 中其他方法以使用這個重新命名的屬性
- 從 NavbarEl 的登入和登出方法中移除控制台日志
- 從 NavbarEl 的登出方法中移除一個不必要的 try-catch 塊
- 在 LoginView 中，導入並使用成員存儲進行狀態管理
- 移除 LoginView 中的控制台日志並重構登入成功處理器，以解構並直接使用回應資料
- 在 LoginView 中成功登入後使用成員存儲設置成員狀態和資料

請記得翻譯提供的所有 git 提交信息。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat: 精緻化註冊流程與介面

- 將註冊封面圖片的 PNG 替換為 SVG
- 更新註冊視圖的網格格式及圖片展示
- 新增用戶同意隱私政策和條款的表單欄位
- 實現表單提交按鈕的載入狀態
- 在 localStorage 而非 cookies 中儲存授權令牌
- 註冊成功後新增導航至首頁的功能
- 改善表單驗證的錯誤訊息展示

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息:

功能: 重構認證系統以使用基於 Cookie 的令牌

- 引進用於 Cookies 的實用功能，包含獲取、設置和刪除 Cookie 的函數。
- 使用新的實用函數將基於本地存儲的令牌管理替換為基於 Cookie 的令牌管理系統。
- 重構認證檢查以跨多個文件非同步進行，包括 `App.vue`、`LoginView.vue`、`SignupView.vue` 和 `DashboardView.vue`。
- 調整 `LoginView.vue` 和 `SignupView.vue` 的異常處理，僅在有 HTTP 錯誤響應時顯示警告。
- 修改 `DashboardView.vue` 以使用 Store 的登入狀態和觀察者，並移除組件創建鉤子中的直接令牌處理。

Signed-off-by: huan5678 <huan5678@gmail.com>

eee317 commented:
我沒有問題，看了之後發現我們應該之前要討論一下共用的狀態 XDD
感謝把登入狀態加入 pinia
不過這樣我那邊的 navbar 是不是到時候要拔掉那段的程式碼？XD

huan5678 committed:
[fix] dashboard view
